concepts/50 React Components/40 Component Configuration Syntax/55 Markup Customization/5 Using a Custom Component.md

```diff
@@ -1,24 +1,89 @@
-You can define the template markup in a separate component. We recommend using <a href="https://reactjs.org/docs/react-api.html#reactpurecomponent" target="_blank">`React.PureComponent`</a> because `React.Component` can be re-rendered unnecessarily. Alternatively, you can implement the <a href="https://reactjs.org/docs/react-component.html#shouldcomponentupdate" target="_blank">shouldComponentUpdate()</a> method.
+You can define custom content in a separate component if you need to utilize state or hooks.
 
-In the following code, custom components are used to specify the [List](https://js.devexpress.com/Demos/WidgetsGallery/Demo/List/ListEditingAndAPI/React/Light)'s [itemTemplate](/api-reference/10%20UI%20Components/dxList/1%20Configuration/itemTemplate.md '/Documentation/ApiReference/UI_Components/dxList/Configuration/#itemTemplate') and the [Button](https://js.devexpress.com/Demos/WidgetsGallery/Demo/Button/PredefinedTypes/React/Light)'s [template](/api-reference/10%20UI%20Components/dxButton/1%20Configuration/template.md '/Documentation/ApiReference/UI_Components/dxButton/Configuration/#template'). Template variables are passed to the components as props.
+[note] For Class components, we recommend using <a href="https://react.dev/reference/react/PureComponent" target="_blank">`React.PureComponent`</a> because `React.Component` can be re-rendered unnecessarily. Alternatively, you can implement the <a href="https://react.dev/reference/react/Component#shouldcomponentupdate" target="_blank">shouldComponentUpdate()</a> method.
+
+The following code demonstrates a [List](/Documentation/Guide/UI_Components/List/Getting_Started_with_List/) which items are rendered in another component. This component is passed to the [itemComponent](/Documentation/ApiReference/UI_Components/dxList/Configuration/#itemComponent) property. 
```

Comment:
A bit weird statement: "a List wich items are rendered in another component". For me it sounds like the list items are removed from the List and rendered inside another 'master' component (another List or so). But it's not the case. Here I would say that to render List items we create a standalone ListItem component. If required, the ListItem component can be reused in another List.